util/fibers/detail/scheduler.cc

```diff
@@ -484,6 +492,24 @@ void Scheduler::PrintAllFiberStackTraces() {
   ExecuteOnAllFiberStacks(print_fn);
 }
 
+bool Scheduler::RunWorkerFibersStepImpl() {
+  DCHECK(!ready_queue_.empty());
+
+  FiberInterface* fi = PopReady();
+  DCHECK(!fi->list_hook.is_linked());
+  DCHECK(!fi->sleep_hook.is_linked());
+  AddReady(dispatch_cntx_.get());
+
+  DVLOG(2) << "Switching to " << fi->name();
+  ProactorBase::UpdateMonotonicTime();
+  fi->SwitchTo();
+
+  was_yield_ = false;
```

Comment:
Should be true right ? We yieled, we switched to a different fiber

Reply:
Oh nvm, we set it to true when we actually yield. Got it.

Reply:
no, `was_yield_` is set to true  by a fiber that yields. here we reset it back. will clarify in the next PRs